actors/miner/src/lib.rs

```diff
@@ -3848,6 +4413,64 @@ where
     Ok(())
 }
 
+fn validate_expiration2(
+    policy: &Policy,
+    curr_epoch: ChainEpoch,
+    activation: ChainEpoch,
+    expiration: ChainEpoch,
+    seal_proof: RegisteredSealProof,
+) -> Result<(), ActorError> {
+    // Expiration must be after activation. Check this explicitly to avoid an underflow below.
+    if expiration <= activation {
+        return Err(actor_error!(
+            illegal_argument,
+            "sector expiration {} must be after activation {}",
+            expiration,
+            activation
+        ));
+    }
+
+    // expiration cannot be less than minimum after activation
+    if expiration - activation < policy.min_sector_expiration {
+        return Err(actor_error!(
+            illegal_argument,
+            "invalid expiration {}, total sector lifetime ({}) must exceed {} after activation {}",
+            expiration,
+            expiration - activation,
+            policy.min_sector_expiration,
+            activation
+        ));
+    }
+
+    // expiration cannot exceed MaxSectorExpirationExtension from now
+    if expiration > curr_epoch + policy.max_sector_expiration_extension {
+        return Err(actor_error!(
+            illegal_argument,
+            "invalid expiration {}, cannot be more than {} past current epoch {}",
+            expiration,
+            policy.max_sector_expiration_extension,
+            curr_epoch
+        ));
+    }
+
+    // total sector lifetime cannot exceed SectorMaximumLifetime for the sector's seal proof
+    let max_lifetime = seal_proof_sector_maximum_lifetime(seal_proof).ok_or_else(|| {
+        actor_error!(illegal_argument, "unrecognized seal proof type {:?}", seal_proof)
+    })?;
+    if expiration - activation > max_lifetime {
+        return Err(actor_error!(
+        illegal_argument,
+        "invalid expiration {}, total sector lifetime ({}) cannot exceed {} after activation {}",
+        expiration,
+        expiration - activation,
+        max_lifetime,
+        activation
+    ));
+    }
+
+    Ok(())
+}
+
 /// Check expiry is exactly *the epoch before* the start of a proving period.
 fn validate_expiration<BS, RT>(
```

Comment:
This appears to be unused now, so delete it and drop the `2` on the method above